### PR TITLE
Make formatting test fail more clear

### LINKED ISSUE #1238: Make our "formatting" test fail more clearly
When a test fails because the code has incorrect formatting, it is currently not very clear. The latest example of this is: https://github.com/symengine/symengine/pull/1228#issuecomment-289468690, the error is buried in the middle of the output (https://travis-ci.org/symengine/symengine/jobs/214969418#L736-L759).

Rather, we should make it prominent. Perhaps create a dedicated Travis test just for this, and at the end somehow visually make it clear that the formatting either passed, or failed (and provide the patch that fixes it, as we do already).

### PR DESCRIPTION
**Pull request by ranjithkumar007**
possible fix for #1238 , going by the idea of a dedicated travis test for checking formatting as suggested by @certik in the issue.

Fixes #1238

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -66,6 +77,11 @@ matrix:
           - binutils-dev
           - g++-5
           - clang-format-3.7
+
+      install:
+        - source bin/install_travis.sh
```
Comment (isuruf) [inline .travis.yml]: There's no need to install cmake, gmp, etc right?

File: .travis.yml [hunk 2]
```diff
@@ -66,6 +77,11 @@ matrix:
           - binutils-dev
           - g++-5
```
Comment (isuruf) [inline .travis.yml]: You can remove the above two packages and `ubuntu-toolchain-r-test`

File: .travis.yml [hunk 3]
```diff
@@ -54,7 +54,18 @@ matrix:
...
   include:
-    - env: BUILD_TYPE="Debug" WITH_BFD="yes" WITH_LATEST_GCC="yes" TEST_CLANG_FORMAT="yes" WITH_COVERAGE="yes" WITH_MPC="yes"
+    - env: BUILD_TYPE="Debug" WITH_BFD="yes" WITH_LATEST_GCC="yes" WITH_COVERAGE="yes" WITH_MPC="yes"
+      compiler: gcc
+      os: linux
+      addons:
+        apt:
+          sources:
+          - ubuntu-toolchain-r-test
+          - llvm-toolchain-precise-3.7
```
Comment (isuruf) [inline .travis.yml]: This is not needed

File: .travis.yml [hunk 4]
```diff
@@ -54,7 +54,18 @@ matrix:
...
+          packages:
+          - binutils-dev
+          - g++-5
+    - env: BUILD_TYPE="Debug" WITH_BFD="yes" TEST_CLANG_FORMAT="yes"
```
Comment (isuruf) [inline .travis.yml]: `BUILD_TYPE="Debug" WITH_BFD="yes" ` is not needed right?

Comment (ranjithkumar007) [inline .travis.yml]: yes,they are not needed. thanks

File: .travis.yml (hunk 3 above)
Comment (ranjithkumar007) [inline .travis.yml]: removing any one from these sources is resulting in error , See [1](https://travis-ci.org/ranjithkumar007/symengine/jobs/216311941/config) , [2](https://travis-ci.org/ranjithkumar007/symengine/jobs/216292580/config).

Comment (isuruf) [inline .travis.yml]: yes, this is fine

File: .travis.yml [hunk 5]
```diff
@@ -54,18 +54,30 @@ matrix:
...
+      os: linux
+      addons:
+        apt:
+          sources:
+          - ubuntu-toolchain-r-test
+          - llvm-toolchain-precise-3.7
+          packages:
           - clang-format-3.7
+      install:
+        - source bin/install_travis.sh
```
Comment (isuruf) [inline .travis.yml]: Instead of running this script, run something like `echo "nothing to install"`

**Comment by ranjithkumar007** [comment]
ping @isuruf , @certik .

**Pull request merged**